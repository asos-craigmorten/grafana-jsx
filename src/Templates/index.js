import PropTypes from "prop-types";

const Templates = ({ children, enable = true, ...props }) => (
  <templating enable={enable} list={children} {...props} />
);

Templates.PropTypes = {
  children: PropTypes.arrayOf(PropTypes.object),
  enable: PropTypes.bool,
};

export default Templates;
