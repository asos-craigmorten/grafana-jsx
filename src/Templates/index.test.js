import Templates from "./";

describe("Templates", () => {
  describe("when used in isolation", () => {
    describe("when passed no children", () => {
      it("should render a templating object containing an empty list", () => {
        expect(<Templates />).toMatchSnapshot();
      });
    });

    describe("when passed children", () => {
      it("should render a templating object containing a template", () => {
        expect(
          <Templates>
            <test-template test-template-prop={"test-template-value"} />
            <test-template test-template-prop={"test-template-value"} />
          </Templates>
        ).toMatchSnapshot();
      });
    });
  });

  describe("when used within another object", () => {
    describe("when passed no children", () => {
      it("should add a templating object containing an empty list to the outer object", () => {
        expect(
          <test>
            <Templates />
          </test>
        ).toMatchSnapshot();
      });
    });

    describe("when passed children", () => {
      it("should add a templating object containing a template to the outer object", () => {
        expect(
          <test>
            <Templates>
              <test-template test-template-prop={"test-template-value"} />
              <test-template test-template-prop={"test-template-value"} />
            </Templates>
          </test>
        ).toMatchSnapshot();
      });
    });
  });
});
