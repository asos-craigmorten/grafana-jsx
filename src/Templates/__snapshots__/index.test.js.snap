// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates when used in isolation when passed children should render a templating object containing a template 1`] = `
Object {
  "enable": true,
  "list": Array [
    Object {
      "test-template-prop": "test-template-value",
    },
    Object {
      "test-template-prop": "test-template-value",
    },
  ],
}
`;

exports[`Templates when used in isolation when passed no children should render a templating object containing an empty list 1`] = `
Object {
  "enable": true,
  "list": Array [],
}
`;

exports[`Templates when used within another object when passed children should add a templating object containing a template to the outer object 1`] = `
Object {
  "templating": Object {
    "enable": true,
    "list": Array [
      Object {
        "test-template-prop": "test-template-value",
      },
      Object {
        "test-template-prop": "test-template-value",
      },
    ],
  },
}
`;

exports[`Templates when used within another object when passed no children should add a templating object containing an empty list to the outer object 1`] = `
Object {
  "templating": Object {
    "enable": true,
    "list": Array [],
  },
}
`;
